packit/utils/changelog_helper.py

```diff
@@ -70,20 +70,38 @@ def update_dist_git(self, full_version: str, upstream_tag: str) -> None:
                 self.dg.get_absolute_specfile_path(),
             )
 
-    def prepare_upstream_using_source_git(self) -> None:
-        """
-        Updates changelog when creating SRPM within source-git repository.
-        """
+    def _get_release_for_source_git(
+        self, current_commit: str, bump_version: bool, release_suffix: Optional[str]
+    ) -> str:
         old_release = self.up.specfile.get_release_number()
+        if release_suffix:
+            return f"{old_release}.{release_suffix}"
+
+        if not bump_version:
+            return old_release
+
         try:
             old_release_int = int(old_release)
             new_release = str(old_release_int + 1)
         except ValueError:
             new_release = str(old_release)
 
+        return f"{new_release}.g{current_commit}"
+
+    def prepare_upstream_using_source_git(
+        self, bump_version: bool, release_suffix: Optional[str]
+    ) -> None:
+        """
+        Updates changelog when creating SRPM within source-git repository.
+        """
         current_commit = self.up.local_project.commit_hexsha
-        release_to_update = f"{new_release}.g{current_commit}"
-        msg = self.entry_from_action or f"- Downstream changes ({current_commit})"
+        release_to_update = self._get_release_for_source_git(
+            current_commit, bump_version, release_suffix
+        )
+
+        msg = self.entry_from_action
+        if not msg and bump_version:
+            msg = f"- Downstream changes ({current_commit})"
```

Comment:
I'm wondering about this: should we put a new changelog entry when the version/release is not being changed?

I know that some packagers work like that - they put new lines in the changelog for a future release bump to keep track of changes.

Reply:
Not sure if it makes sense in this case, since we're not putting commit messages, just non-descriptive `Downstream changes {hash}` :thinking:

Reply:
True, so let's not touch it then :)